### PR TITLE
Add docs to more stdlib exceptions

### PR DESCRIPTION
**Pull request by whatyouhide**
Like #12735, but for more exceptions.

### CODE REVIEW

File: lib/elixir/lib/exception.ex [hunk 1]
```diff
@@ -1670,6 +1739,17 @@ defmodule File.Error do
...
 
 defmodule File.CopyError do
+  @moduledoc """
+  An exception that is raised when copying a file fails.
+
+  The following fields of this exception are public and can be accessed freely:
+
+    * `:source` (`t:Path.t/0`) - the source path
+    * `:destination` (`t:Path.t/0`) - the destination path
+    * `:reason` (`t:term/0`) - the reason for the error
```
Comment (viniciusmuller) [inline lib/elixir/lib/exception.ex:1749]:
```suggestion
    * `:reason` (`t:term/0`) - the reason why the file could not be copied
```

File: lib/elixir/lib/exception.ex [hunk 2]
```diff
@@ -1688,6 +1768,17 @@ defmodule File.CopyError do
...
 
 defmodule File.RenameError do
+  @moduledoc """
+  An exception that is raised when renaming a file fails.
+
+  The following fields of this exception are public and can be accessed freely:
+
+    * `:source` (`t:Path.t/0`) - the source path
+    * `:destination` (`t:Path.t/0`) - the destination path
+    * `:reason` (`t:term/0`) - the reason for the error
```
Comment (viniciusmuller) [inline lib/elixir/lib/exception.ex:1778]:
```suggestion
    * `:reason` (`t:term/0`) - the reason why the file could not be renamed
```

File: lib/elixir/lib/exception.ex [hunk 3]
```diff
@@ -1706,6 +1797,17 @@ defmodule File.RenameError do
...
 
 defmodule File.LinkError do
+  @moduledoc """
+  An exception that is raised when linking a file fails.
+
+  The following fields of this exception are public and can be accessed freely:
+
+    * `:existing` (`t:Path.t/0`) - the existing file to link
+    * `:new` (`t:Path.t/0`) - the link destination
+    * `:reason` (`t:term/0`) - the reason for the error
```
Comment (viniciusmuller) [inline lib/elixir/lib/exception.ex:1807]:
```suggestion
    * `:reason` (`t:term/0`) - the reason why the file could not be linked
```

**Review by viniciusmuller — APPROVED** [review]
LGTM! I've suggested a few changes in some `:reason` fields in order to stay more consistent with how it was worded in `Code.LoadError`. feel free to apply them if you think it makes sense :)

**Review by josevalim — APPROVED** [review]

**Review by sabiwara — APPROVED** [review]
LGTM!

**Pull request merged**